pandas/core/generic.py

```diff
@@ -7148,12 +7148,18 @@ def _clip_with_scalar(self, lower, upper, inplace=False):
             raise ValueError("Cannot use an NA value as a clip threshold")
 
         result = self
-        if upper is not None:
-            subset = self.le(upper, axis=None) | isna(result)
-            result = result.where(subset, upper, axis=None, inplace=False)
-        if lower is not None:
-            subset = self.ge(lower, axis=None) | isna(result)
-            result = result.where(subset, lower, axis=None, inplace=False)
+        mask = isna(self.values)
+
+        with np.errstate(all='ignore'):
+            if upper is not None:
+                subset = self.values <= upper
```

Comment:
Curious, does this place nice when self is the new `Int64` dtype?

Reply:
It appears to - let me know if there's a more extensive test you have in mind:
```
In [2]: s = pd.Series(range(5)).astype('Int64')
In [3]: s.clip(1, 3)
Out[3]: 0    1
        1    1
        2    2
        3    3
        4    3
        dtype: Int64
```